### PR TITLE
[NDS-990] feat: table sorting

### PR DESCRIPTION
**Pull request by kostasdano**
## Description

[NDS-990](https://orfium.atlassian.net/browse/NDS-990)

[NDS-990]: https://orfium.atlassian.net/browse/NDS-990?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: src/components/Table/Table.stories.tsx [hunk 1]
```diff
@@ -105,3 +108,79 @@ export const ColumnChooser = {
     },
   },
 };
+
+export const Sorting = {
+  render: (args) => {
+    const { rowSize, isAlwaysVisible = [], isMultiSortable } = args;
+
+    const [sorting, setSorting] = useState<SortingState>();
```
Comment (panvourtsis) [inline src/components/Table/Table.stories.tsx:116]: can we have multiple sorting examples with the isMultiple or document it well would work as well.

File: src/components/Table/Table.mdx [hunk 2]
```diff
@@ -49,3 +49,7 @@ This type includes all the configuration for the columns
 ### Column Chooser
 
 <Canvas of={TableStories.ColumnChooser} />
+
+### Sorting
+
```
Comment (panvourtsis) [inline src/components/Table/Table.mdx:54]: missing documentation on how to use it, multiple vs single, buttons explaining

File: src/components/Table/Table.stories.tsx (hunk 1 above)
Comment (kostasdano) [reply src/components/Table/Table.stories.tsx:116]: check latest commit for documentation

File: src/components/Table/Table.mdx (hunk 2 above)
Comment (kostasdano) [reply src/components/Table/Table.mdx:54]: check latest commit for documentation

**Comment by kostasdano** [comment]
> Keyboard navigation doesnt reveal the buttons :). I would expect on hover/focus to reveal them Screenshot 2024-05-08 at 11 25 12 AM

I revisited the keyboard navigation for all table actions, you can check on the review app

**Comment by panvourtsis** [comment]
image
https://designlab-pr-806.orfium-dev.com/?path=/story/updated-components-table-table--sorting

@kostasdano Sorting table doesnt affect the data

**Review by panvourtsis — APPROVED** [review]

**Pull request merged**